### PR TITLE
i#1312 AVX-512 support: Re-factor OPCODE_FOR_CREATE in api.ir.

### PR DESCRIPTION
**Pull request by hgreving2304**
As part of adding more tests for AVX-512, this minor re-factors to re-using
OPCODE_FOR_CREATE macros for 0, 1, 2, 3, 4, .. args.

Issue: #1312

### CODE REVIEW

File: suite/tests/api/ir_x86.c [hunk 1]
```diff
@@ -222,28 +236,75 @@ test_all_opcodes_3(void *dc)
...
+{
+#    define INCLUDE_NAME "ir_x86_4args.h"
+#    include "ir_x86_all_opc.h"
 #    undef INCLUDE_NAME
 }
 
+#    undef OPCODE_FOR_CREATE
+#    undef XOPCODE_FOR_CREATE
+
+/***************************************************************************/
```
Comment (derekbruening) [inline suite/tests/api/ir_x86.c]: Use the original trailing banner style which this PR deleted: it indicates closing while this flat line does not.

File: suite/tests/api/ir_x86.c [hunk 2]
```diff
@@ -111,10 +111,9 @@ static byte buf[8192];
 #define X86_ONLY 1
 #define X64_ONLY 2
 
-static void
-test_all_opcodes_0(void *dc)
-{
-#define INCLUDE_NAME "ir_x86_0args.h"
+/***************************************************************************/
+/*********************** OPCODE_FOR_CREATE 0 args **************************/
```
Comment (derekbruening) [inline suite/tests/api/ir_x86.c]: Use the banner style in the style guide, here and below.

File: suite/tests/api/ir_x86.c [hunk 3]
```diff
@@ -222,28 +236,75 @@ test_all_opcodes_3(void *dc)
...
+static void
+test_all_opcodes_3_avx512_vex(void *dc)
+{
+#    define INCLUDE_NAME "ir_x86_3args_avx512_vex.h"
+#    include "ir_x86_all_opc.h"
+#    undef INCLUDE_NAME
+}
+
+static void
+test_all_opcodes_mask_3_avx512_evex(void *dc)
```
Comment (derekbruening) [inline suite/tests/api/ir_x86.c]: A more consistent name would be test_all_opcodes_3_avx512_evex_mask() and "ir_x86_3args_avx512_evex_mask.h" (or swapping evex and mask: mainly putting the num args before avx512 to match all the other names).

File: suite/tests/api/ir_x86.c (hunk 1 above)
Comment (hgreving2304) [reply suite/tests/api/ir_x86.c]: Done.

File: suite/tests/api/ir_x86.c (hunk 2 above)
Comment (hgreving2304) [reply suite/tests/api/ir_x86.c]: Done.

File: suite/tests/api/ir_x86.c (hunk 3 above)
Comment (hgreving2304) [reply suite/tests/api/ir_x86.c]: Done.

**Pull request merged**